File: src/main/java/com/iri/training/web/service/CommentService.java
package com.iri.training.web.service;

import java.sql.SQLException;
import java.util.List;

import com.iri.training.enums.SubjectType;
import com.iri.training.model.Comment;

public interface CommentService {

	Comment getCommentById(long commentId, boolean getComments) throws SQLException;

	List<Comment> getCommentsBySubject(SubjectType subjectType, long subjectId, boolean getComments) throws SQLException;

	void addComment(Comment comment) throws SQLException;

	void deleteComment(long commentId) throws  SQLException;

	void deleteCommentReplies(SubjectType subjectType, long parentId) throws SQLException;

	void editComment(Comment comment) throws SQLException;
}

File: src/main/webapp/app/core/comment/comment.service.js
'use strict';

angular.
    module('core.comment').
    factory('Comment', ['$resource',
        function($resource) {
            return {
                Comment: $resource('api/comment/:commentId', {
                    params: {
                        getComments:'@getComments'
                    }
                }),
                CommentsBySubject: $resource('api/comment/subject/:subjectType/:subjectId', {
                    params: {
                        getComments:'@getComments'
                    }
                }),
                AddComment: $resource('api/comment/add'),
                DeleteComment: $resource('api/comment/:commentId/delete'),
                EditComment: $resource('api/comment/edit')
            };
        }
    ]);
